MDGameSynchronizer/MDGameSynchronizer.cs

```diff
@@ -297,470 +302,468 @@ private void PauseGame()
-            // This is the first time we synched all nodes, notify the server
-            MDLog.Debug(LOG_CAT, "Node synch complete, notifying server");
-            NodeSynchCompleted = true;
-            RpcId(GameSession.GetNetworkMaster(), nameof(NotifyAllNodesSynched));
-        }
+            // Only do this on clients
+            if (!this.IsClient())
+            {
+                timer.RemoveAndFree();
+                return;
+            }
 
-        int NotSynchedNodes = 0;
+            if (NodeCount < 0)
+            {
+                MDLog.Debug(LOG_CAT, "We got no node count");
+                // We don't know how many nodes we got yet
+                return;
+            }
 
-        // Check node custom logic to see if synch is done
-        foreach (Node node in NodeList)
-        {
-            if (node is IMDSynchronizedNode)
+            int SynchedNodes = NodeCount;
+            if (NodeList.Count < NodeCount)
+            {
+                MDLog.Debug(LOG_CAT, "We still don't have all nodes");
+                // We still don't have all nodes
+                SynchedNodes = NodeList.Count;
+            }
+            else if (!NodeSynchCompleted)
+            {
+                // This is the first time we synched all nodes, notify the server
+                MDLog.Debug(LOG_CAT, "Node synch complete, notifying server");
+                NodeSynchCompleted = true;
+                RpcId(GameSession.GetNetworkMaster(), nameof(NotifyAllNodesSynched));
+            }
+
+            int NotSynchedNodes = 0;
+
+            // Check node custom logic to see if synch is done
+            foreach (Node node in NodeList)
             {
-                if (!((IMDSynchronizedNode)node).IsSynchronizationComplete())
+                if (!(node is IMDSynchronizedNode) || ((IMDSynchronizedNode) node).IsSynchronizationComplete())
                 {
-                    // We are not synched
-                    MDLog.Trace(LOG_CAT, "A node is still synching: {0}", node.GetPath());
-                    SynchedNodes--;
-                    NotSynchedNodes++;
+                    continue;
                 }
+
+                // We are not synched
+                MDLog.Trace(LOG_CAT, "A node is still synching: {0}", node.GetPath());
+                SynchedNodes--;
+                NotSynchedNodes++;
+            }
+
+            if (SynchedNodes == NodeCount)
+            {
+                // We are done synching
+                RpcId(MDStatics.GetServerId(), nameof(ClientSynchDone));
+                MDLog.Debug(LOG_CAT, "We are done synching notifying server");
+                // Set ourselves to done
+                OnPlayerSynchStatusUpdateEvent(MDStatics.GetPeerId(), 1f);
+                timer.RemoveAndFree();
+            }
+            else
+            {
+                float percentage = (float) SynchedNodes / NodeCount;
+                // TODO: Maybe using this instead of formatting string would be better?
```

Comment:
Forgot this was a thing, yeah I think we should use string interpolation whenever we can.

Reply:
I've made this into string interpolation everywhere where i could find.